modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicPostPreFilterSplitter.java

```diff
@@ -49,6 +49,8 @@ public VectorMosaicPostPreFilterSplitter(SimpleFeatureType featureType) {
         indexAttributeDescriptors.stream()
                 .map(AttributeDescriptor::getLocalName)
                 .forEach(this.attributeNames::add);
+        // add the "default geometry" attribute
+        this.attributeNames.add("");
```

Comment:
Maybe add a comment with a See link to DefaultGeometryFeaturePropertyAccessor ?  This should probably be defined somewhere as a static constant somewhere to avoid confusion, but probably not in this PR.